ui/app/src/components/CreateDashboardDialog/CreateDashboardDialog.tsx

```diff
@@ -0,0 +1,108 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+
+export interface CreateDashboardDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onSubmit: Dispatch<string>;
+  projectName: string;
+}
+
+/**
+ * Dialog used to create a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.projectName The project where the dashboard will be created.
+ * @constructor
+ */
+const CreateDashboardDialog = (props: CreateDashboardDialogProps) => {
+  const { open, onClose, onSubmit, projectName } = props;
+  const [name, setName] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    if (!e.target.value) {
+      setError('Required');
+    } else if (!/^[a-zA-Z0-9_.:-]+$/.exec(e.target.value)) {
+      setError('Allowed special characters are _ . : -');
+    } else {
+      setError(undefined);
+    }
+  };
+
+  // Reinitialize form for next time the dialog is opened
+  const resetForm = () => {
+    setName(undefined);
+    setError(undefined);
+  };
+
+  const handleConfirm = () => {
+    if (name) {
+      onClose();
+      onSubmit(name);
+      resetForm();
+    }
+  };
+
+  const handleClose = () => {
+    onClose();
+    resetForm();
+  };
+
+  return (
+    <Dialog open={open} onClose={handleClose} aria-labelledby="confirm-dialog">
+      <DialogTitle id="confirm-dialog">Create a new dashboard in the project {projectName}</DialogTitle>
+      <IconButton
+        aria-label="Close"
+        onClick={() => onClose()}
+        sx={(theme) => ({
+          position: 'absolute',
+          top: theme.spacing(0.5),
+          right: theme.spacing(0.5),
+        })}
+      >
+        <CloseIcon />
+      </IconButton>
+      <DialogContent dividers sx={{ width: '500px' }}>
+        <TextField
+          required
+          margin="dense"
+          id="name"
+          label="Dashboard name"
+          type="text"
+          fullWidth
+          onChange={handleChange}
+          value={name}
+          error={!!error}
+          helperText={error}
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button variant="contained" disabled={!!error} onClick={handleConfirm}>
+          Add
+        </Button>
+        <Button variant="contained" color="secondary" onClick={handleClose}>
```

Comment:
Can `variant="outlined"` be used instead for the 'Cancel' button? That way it's similar to how it looks in other places ([1](https://github.com/perses/perses/blob/main/ui/dashboards/src/components/DiscardChangesConfirmationDialog/DiscardChangesConfirmationDialog.tsx#L46-L48), [2](https://github.com/perses/perses/blob/56dc901aec9895332c7354609512c3748e8bcd64/ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx#L93-L95))

Reply:
Yes, sure